single_kernel_mongo/managers/mongodb_operator.py

```diff
@@ -1019,29 +1046,37 @@ def get_statuses(self, scope: DPHScope, recompute: bool = False) -> list[StatusO
         if not recompute:
             return self.state.statuses.get(scope=scope, component=self.name).root
 
-        if scope == "app":
-            return charm_statuses
+        if scope == "unit" and not self.workload.workload_present:
+            return [CharmStatuses.MONGODB_NOT_INSTALLED.value]
```

Comment:
to be clear, this should never run - right? As we should be error'ing the install hook should it fail to install, so no subsequent hook will ever display the status

Reply:
On VM this should never run.
On K8s, it will happen if the container is not available.